controllers/rhmi/bootstrapReconciler_test.go

```diff
@@ -247,6 +247,121 @@ func TestReconciler_reconcilePrometheusRules(t *testing.T) {
 	}
 }
 
+func TestReconciler_createAutoscalingConfigMap(t *testing.T) {
+	scheme := runtime.NewScheme()
+	corev1.AddToScheme(scheme)
+
+	tests := []struct {
+		Name           string
+		ExpectedStatus integreatlyv1alpha1.StatusPhase
+		FakeConfig     *config.ConfigReadWriterMock
+		FakeMPM        *marketplace.MarketplaceInterfaceMock
+		Installation   *integreatlyv1alpha1.RHMI
+		Recorder       record.EventRecorder
+		InitialObjs    []runtime.Object
+		Assertion      func(k8sclient.Client) error
+	}{
+		{
+			Name:           "Test autoscaling configmap isn't reconciled when it already exists",
+			ExpectedStatus: integreatlyv1alpha1.PhaseCompleted,
+			FakeConfig: &config.ConfigReadWriterMock{
+				GetOperatorNamespaceFunc: func() string {
+					return rhoamOperatorNs
+				},
+			},
+			FakeMPM: &marketplace.MarketplaceInterfaceMock{},
+			Installation: &integreatlyv1alpha1.RHMI{
+				ObjectMeta: v1.ObjectMeta{
+					Namespace: rhoamOperatorNs,
+				},
+				Spec: integreatlyv1alpha1.RHMISpec{
+					NamespacePrefix: "redhat-rhoam-",
+				},
+			},
+			Recorder: record.NewFakeRecorder(50),
+			InitialObjs: []runtime.Object{
+				&corev1.ConfigMap{
+					ObjectMeta: v1.ObjectMeta{
+						Name:      "autoscaling-config",
+						Namespace: rhoamOperatorNs,
+					},
+					Data: map[string]string{"test": "test"},
+				},
+			},
+			Assertion: assertConfigMapNotChanged,
+		},
+		{
+			Name:           "Test autoscaling configmap is created when it is doesn't exist",
+			ExpectedStatus: integreatlyv1alpha1.PhaseCompleted,
+			FakeConfig: &config.ConfigReadWriterMock{
+				GetOperatorNamespaceFunc: func() string {
+					return rhoamOperatorNs
+				},
+			},
+			FakeMPM: &marketplace.MarketplaceInterfaceMock{},
+			Installation: &integreatlyv1alpha1.RHMI{
+				ObjectMeta: v1.ObjectMeta{
+					Namespace: rhoamOperatorNs,
+				},
+				Spec: integreatlyv1alpha1.RHMISpec{
+					NamespacePrefix: "redhat-rhoam-",
+				},
+			},
+			Recorder: record.NewFakeRecorder(50),
+			InitialObjs: []runtime.Object{
+				&corev1.ConfigMap{},
+			},
+			Assertion: assertConfigMapCreated,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.Name, func(t *testing.T) {
+			reconciler, err := NewBootstrapReconciler(tt.FakeConfig, tt.Installation, tt.FakeMPM, tt.Recorder, l.NewLogger())
```

Comment:
There should be a defer action here to set the cluster back to the state it was in before this test started. If not these test could have a side effect on other tests.